wgengine/wgengine.go

```diff
@@ -22,20 +26,29 @@ type PeerStatus struct {
 	NodeKey          tailcfg.NodeKey
 }
 
+// Status is the Engine status.
 type Status struct {
 	Peers      []PeerStatus
 	LocalAddrs []string // TODO(crawshaw): []wgcfg.Endpoint?
 }
 
-type StatusCallback func(s *Status, err error)
+// StatusCallback is the type of status callbacks used by
+// Engine.SetStatusCallback.
+//
+// Exactly one of Status or error is non-nil.
+type StatusCallback func(*Status, error)
 
+// RouteSettings is the full WireGuard config data (set of peers keys,
+// IP, etc in wgcfg.Config) plus the things that WireGuard doesn't do
+// itself, like DNS stuff.
 type RouteSettings struct {
-	LocalAddr  wgcfg.CIDR
+	LocalAddr  wgcfg.CIDR // TODO: why is this here? how does it differ from wgcfg.Config's info?
```

Comment:
On closer inspection, I don't get it. This looks like an oversimplification in the router logic. Implementations of SetRoutes should be using the set of localaddrs in `Cfg.Interface.Addresses`. (It's a fine simplification right now because we give every node a single cgnat address, but we want the option to do more than one later, e.g. ipv6.)

So I think this needs a proper TODO in the code to replace with Cfg.Interface.Addresses.